app/spdb/scripts/spdb

```diff
@@ -0,0 +1,256 @@
+#!/bin/bash
+# 
+# This script manages the startup / shutdown of the ODB.
+
+# RETVALS:
+# 0. Expected.
+# 1. Usage displayed.
+# 2. Could not start.
+# 3. Could not stop.
+# 4. Status: not running.
+
+# Source function library.
+. /etc/rc.d/init.d/functions
+
+# psjava access.
+export PATH=$PATH:/home/software/bin
+
+# System variables.
+SERVER=`hostname`
+PORT=8224
+
+
+# Display usage information.
+usage ()
+{
+    echo $"Usage: $0 {start|stop|status|restart}" 1>&2
+    RETVAL=1
+}
+
+# Sleep for 10 seconds, and display a . after each two second interval.
+sleep_dots ()
+{
+    for i in `seq 1 5`; do
+	printf " ."
+	sleep 2
+    done
+}
+
+# Find the latest version of the ODB.
+# If one is found:
+# 1. SPDB_DIR is set to the path of the spdb directory.
+# 2. SPDB_VERSION is set to the version string (e.g. 1.2.3).
+# 4. SPDB_EXEC is set to the path to the binary.
+determine_spdb_directory ()
+{
+    local SPDB_DIRS=($(ls -d /home/software/spdb*))
+    unset SPDB_DIR
+    unset SPDB_LOCKFILE
+    unset SPDB_VERSION_STR
+    SPDB_SEMESTER=0000A
+    SPDB_VERSION=(0 0 0)
+    
+    for CURR_DIR in "${SPDB_DIRS[@]}"
+    do
+	# Extract the version so that we can find the most recent.
+	# We don't use sort on the ls output for this because it will compare by string
+	# instead of by number, and we want to avoid things like 1.10.1 being less than 1.2.1.
+	CURR_SEMESTER=`echo $CURR_DIR | sed "s/^.*spdb_\([^-.]*\).*$/\1/"`
+	if [[ "$CURR_SEMESTER" < "$SPDB_SEMESTER" ]]; then
```

Comment:
Lexicographic string ordering should be correct for semesters.

Reply:
This comment doesn't change anything about the `init.d` script because we'd still need to figure out the current version anyway, but we should do a better job cleaning up old installations.  We really only need the last installation before an update, and only for a few days.  Once a a bit of time has passed we can't easily go back to a previous version anyway.